### PR TITLE
feat(pagination): Add showSearch prop into showSizeChanger

### LINKED ISSUE #30324: Table 的Pagination 上的下拉每页几条记录功能能否改成可同时输入？
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
table 的Pagination ，每页几条的功能虽可以设定下拉显示记录数，但不够灵活，有时用户想要可输入，如：设定每页显示20，40，80或160等，用户有时想输入如50，300等时，就不好操作

### What does the proposed API look like?
希望将此每页几条的功能，改成下拉的同时也可以输入

### PR DESCRIPTION
**Pull request by jacklee814**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #30324 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Pagination added a showSearch prop to control the Select |
| 🇨🇳 Chinese |  Pagination 新增 showSearch prop 控制分頁的輸入框  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 879bad3</samp>

Added search feature to `Pagination` component and removed unused `Select` component. Simplified code by using existing React and Select components.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 879bad3</samp>

*  Add `showSearch` prop to `PaginationProps` interface to enable search functionality in pagination select component ([link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352R56))
*  Define `CustomSelect` component that wraps imported `Select` component and passes `showSearch` and `size` props to it ([link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352R131-R139))
*  Replace `MiddleSelect` and `MiniSelect` components with `CustomSelect` component in `selectComponentClass` prop ([link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352L131-R148), [link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352L9-R26), [link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-a34e1dd1a02f6ee6067986a1242ba2b093bb074eafa1739c050af8d2dd1edc52))
*  Import React as a namespace and Select component and props type from parent directory ([link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352R1), [link](https://github.com/ant-design/ant-design/pull/46070/files?diff=unified&w=0#diff-6ded05884b813a9ee306b10152a2e379ac9c8c724ade740984c1aed485d14352L9-R26))

### CODE REVIEW

File: components/pagination/index.zh-CN.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*WM86SrBC8TsAAA
 | showTotal | 用于显示数据总量和当前数据顺序 | function(total, range) | - |  |
 | simple | 当添加该属性时，显示为简单分页 | boolean | - |  |
 | size | 当为 `small` 时，是小尺寸分页 | `default` \| `small` | `default` |  |
+| showSearch | 页码选择器配置是否可搜索 | boolean | true |  |
```
Comment (afc163) [inline components/pagination/index.zh-CN.md:59]:
```
showSizeChanger={{ showSearch: boolean }}
```

API 这样设计吧。

**Comment by li-jia-nan** [comment]
feature branch

File: components/pagination/index.zh-CN.md (hunk 1 above)
Comment (jacklee814) [reply components/pagination/index.zh-CN.md:59]: showSizeChanger 在 RcPaginationProps 是 boolean 型別，需要在 Pagination 這邊 Omit showSizeChanger 重新定義成物件嗎？

Comment (afc163) [reply components/pagination/index.zh-CN.md:59]: 对，直接修改 rc-pagination

**Comment by jacklee814** [comment]
> feature branch

我把 base branch 換到 feature 了！

File: components/pagination/index.zh-CN.md (hunk 1 above)
Comment (jacklee814) [reply components/pagination/index.zh-CN.md:59]: 調整了！

Comment (afc163) [reply components/pagination/index.zh-CN.md:59]: 我意思是直接提 PR 到 rc-pagniation 上，而不是在 antd 这层覆盖。

**Comment by afc163** [comment]
https://github.com/react-component/pagination/pull/533

**Comment by chenzhihang950909** [comment]
我想知道这个提案到底通过了没？？？

**Pull request closed without merging**